packages/cozy-sharing/src/components/ShareByLink.jsx

```diff
@@ -52,12 +52,13 @@ class ShareByLink extends React.Component {
     }
   }
 
-  copyLinkToClipboard = () => {
+  copyLinkToClipboard = ({ isAutomaticCopy }) => {
     if (copy(this.props.link))
       Alerter.success(
         this.props.t(`${this.props.documentType}.share.shareByLink.copied`)
       )
-    else
+    else if (!isAutomaticCopy)
+      // Copy after a XHR query fails in safari
```

Comment:
I think the behavior of Safari is totally normal, we should not copy after the XHR, it is a security problem and I would not be surprised if browsers in the future prevent us from doing that. To prevent that we could modify the UI so that the copy to clipboard button is added after the link has been created.

Reply:
> To prevent that we could modify the UI so that the copy to clipboard button is added after the link has been created.

It is :). We try to copy the link to the clipboard right away for convenience, but we do indeed add a copy to clipboard button in the UI that works as expected.

Reply:
Ah ok, good. In this case I think the comment would be clearer if it read "In case of automatic copy, the browser can block the copy request. This is not shown to the user since it is expected and can be circumvented by clicking directly on the copy link".

Reply:
From an API point of view, I wonder if it would not be better to have ignoreErrors: true instead of isAutomaticCopy. If you read the call point, you can wonder why is there isAutomaticCopy there.

Reply:
I updated the comment 👍 .

> If you read the call point, you can wonder why is there isAutomaticCopy there.

I wondered about that as well, but then it makes it less clear why the error is ignored in one case and not the other. I think it's better to make it explicit why the error is suppressed rather than why the function is called with an extra arg.

Reply:
I see, thanks for the explanation.